### ISSUE TITLE
Fee setting / estimating / defaults needs an urgent overhaul

### ISSUE THREAD

**Issue by n9jd34x04l151ho4** [fyookball/electrum#246]
I think the software should just default to 1 satoshi per byte. There is no need to pay more than that on Bitcoin Cash. There is plenty of room in the blocks and will be for a long time to come. We can eliminate the slider completely for now for simplicity. Complicated fee estimating mechanisms are for SegShit coin which can't process all the transactions in the mempool each block.

Now as you know, 1 satoshi = 0.00000001 (8 decimal places).

I tried setting that in my Electron Cash and it crashed to the console. Then it wouldn't re-open again:

```
$ Traceback (most recent call last):
  File "/home/c/electron-cash-installation/ElectronCash-2.9.4/electron-cash", line 373, in <module>
    d.init_gui(config, plugins)
  File "/home/c/electron-cash-installation/ElectronCash-2.9.4/lib/daemon.py", line 283, in init_gui
    self.gui.main()
  File "/home/c/electron-cash-installation/ElectronCash-2.9.4/gui/qt/__init__.py", line 227, in main
    if not self.start_new_window(path, self.config.get('url')):
  File "/home/c/electron-cash-installation/ElectronCash-2.9.4/gui/qt/__init__.py", line 193, in start_new_window
    w = self.create_window_for_wallet(wallet)
  File "/home/c/electron-cash-installation/ElectronCash-2.9.4/gui/qt/__init__.py", line 168, in create_window_for_wallet
    w = ElectrumWindow(self, wallet)
  File "/home/c/electron-cash-installation/ElectronCash-2.9.4/gui/qt/main_window.py", line 137, in __init__
    self.send_tab = self.create_send_tab()
  File "/home/c/electron-cash-installation/ElectronCash-2.9.4/gui/qt/main_window.py", line 1064, in create_send_tab
    self.fee_slider = FeeSlider(self, self.config, fee_cb)
  File "/home/c/electron-cash-installation/ElectronCash-2.9.4/gui/qt/fee_slider.py", line 20, in __init__
    self.update()
  File "/home/c/electron-cash-installation/ElectronCash-2.9.4/gui/qt/fee_slider.py", line 47, in update
    pos = min(fee_rate / self.fee_step, 10)
ZeroDivisionError: integer division or modulo by zero
```

I tried to edit the `~/.electron-cash/config` file manually but it wouldn't accept the fee amount no matter what. Just kept crashing. I had to delete the config file completely and then it finally opened again.

Anyway, by default with a fresh config the wallet defaults to showing balance in mBCH e.g. `1,300.33221 mBCH`. I don't know any site that uses that measurement for payments. What's a mBCH anyway, milliBCH? 1.2 BCH == 1,200 mBCH? Too confusing for plebs like me. It would seem simpler just to use BCH as the default at this stage. Maybe when BCH is worth a million dollars it might be worth changing the default to satoshis.

In the preferences window it's even worse, for the static fee, now I have to convert between mBCH and kB. Is that Kilobytes or KibiBytes? Let's use SI notation.

![image](https://user-images.githubusercontent.com/31981309/32703639-cb35e114-c7df-11e7-9f4c-4143bd6af833.png)

Anyway I'll set my Units to BCH for now to avoid confusion:

![image](https://user-images.githubusercontent.com/31981309/32703707-e76712ee-c7e0-11e7-8a02-5e60f7364558.png)

Max static fee is now 1 BCH/kB 

![image](https://user-images.githubusercontent.com/31981309/32704068-c178624e-c7e6-11e7-9838-bade32004ecb.png)

That seems a bit high so let's try setting it to 1 Satoshi a byte. That must be 0.00001 BCH/kB (0.00001 / 1000 == 0.00000001 per byte). Already the text block is not big enough and the text is overlayed:

![image](https://user-images.githubusercontent.com/31981309/32704246-1718552c-c7e9-11e7-83b8-00d935356fcd.png)

Now I will make a payment:

![image](https://user-images.githubusercontent.com/31981309/32704279-7f1b8842-c7e9-11e7-8ca8-39eed75a2af2.png)

The fee text box is not big enough still with text overwriting here as well. If I click Preview I get an error:

![image](https://user-images.githubusercontent.com/31981309/32704287-a270e0f8-c7e9-11e7-893b-b308d63693ab.png)

The network will happily process 1 sat/byte so this limitation needs to be removed.

Dragging the fee slider is kinda useless. Dragging to the left side it says `Fixed rate: 0.1 Sat/B Low fee`. Dragging to the right side the hover says `Fixed rate: 1.0 Sat/B Within 25 blocks`. That's got to be incorrect surely. 1 Sat/Byte will go through next block or two max if it didn't get relayed in time. Also the calculation appears to be 0.00001328 BCH fee to send 0.1 BCH. If the average transaction is 257 bytes why do I have to pay 5 times that? Is it combining inputs perhaps? I can't tell the preview is blocked by the low fee warning.

Anyway I will delete my config again, reload and check the default fees.

![image](https://user-images.githubusercontent.com/31981309/32704400-8695e732-c7eb-11e7-8fd8-83a0f34ecdf1.png)

Fee slider is in the middle, `Fixed rate: 50.0 Sat/B Within 1 blocks`. Ouch. Slider to the left: `10 Sat/B` and slider right: `100.0 Sat/B`. I move to the 10 Sat/B option and click Preview:

![image](https://user-images.githubusercontent.com/31981309/32704452-648c0a8a-c7ec-11e7-82b0-e2923706037e.png)

Now fee is at 10.32 Sat/B. Ouch. Anyway 10.32 x 226 bytes = 0.00002332 BCH or 0.02332 mBCH not 0.02333 mBCH. Someone's getting an extra Satoshi here.

I think the amount sent is 0. mBCH because I am sending to my own address. Interesting in this scenario is that when I send this transaction it only shows the outbound transaction in my wallet. The incoming one is not shown. But after the transaction completes I am only missing the fee from the total balance.

Anyway I think this needs an overhaul. We can simplify this and make it the best, easiest to use desktop wallet for the masses.

**Comment by fyookball** [comment]
we actually removed most of the 'dynamic fee' stuff.  I'll try recreating the division by zero stuff.

**Comment by BitAlien** [comment]
I've also experienced this weird fee estimation. A transaction of 226 bytes, set to 1 satoshi per byte, gives a fee of 270 satoshis.

**Comment by fyookball** [comment]
fees are mostly fixed in 3.1.2
